File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    result = [0]
    
    for i in range(len(numbers)):
        tmp = []
        for j in range(len(result)):
            tmp.append(result[j] + numbers[i])
            tmp.append(result[j] - numbers[i])
        
        result = tmp
    
    answer = result.count(target)
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer = 0;
    
    public int solution(int[] numbers, int target) {  
        
        dfs(0, 0, numbers, target);
        
        return answer;
    }
    
    private void dfs(int k, int curSum, int[] numbers, int target) {
        if (k == numbers.length) {
            if (curSum == target) answer++; 
            return;
        }
        
        int num = numbers[k];
        dfs(k + 1, curSum + num, numbers, target);
        dfs(k + 1, curSum - num, numbers, target);
    }
}